### ISSUE TITLE
New check suggestion: Default/assignment to $1: bash says 'cannot assign in this way'

### ISSUE THREAD

**Issue by JoshuaSCochrane** [koalaman/shellcheck#2183]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

code=${1:=1}

f ()
{
inner=${1:=1}
echo "$inner"
}

echo "$code"
f
```

#### Here's what shellcheck currently says:
```
No issues detected!
```

#### Here's what I wanted or expected to see:

Bash complains:
```
myscript: line 4: $1: cannot assign in this way
myscript: line 7: $1: cannot assign in this way
```

Would be nice for SC to catch this!

**Comment by JoshuaSCochrane** [comment]
Oh, and maybe suggest `${1:-1}` (default-value) here instead, which is what I _meant_ to use ;)

From https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Shell-Parameter-Expansion:

> `${parameter:=word}`
If _parameter_ is unset or null, the expansion of _word_ is assigned to _parameter_. The value of _parameter_ is then substituted. Positional parameters and special parameters may not be assigned to in this way.

So it must apply to 'special parameters' as well, whatever those are.
